### ISSUE TITLE
support customized process filter when using command:PickProcess in node attach to process debugger 

### ISSUE THREAD

**Issue by aprilandjan** [microsoft/vscode#95598]
The vscode debugger provides really convenient configurations when try to debug a node.js process. Recently I'm using `Attach to Process` with `command:PickProcess` and it works perfectly. (see <https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_attach-to-node-process-action>)

The picker command showed all the the other programs that were launched with one of the various forms of `--debug` or `--inspect` arguments. However, most of times users have to select the specific one manually because of the others are possibly just distractions.

If there is some configuration which support customizing process filter for available debuggable process list, then we can directly configure which process to be listed; If only one process is matched, attach to it automatically. This might make the debugger task even more convenient.

Is there any way to approach similar purpose, or can we have this feature in future? Thanks!

**Comment by isidorn** [comment]
The pick process command is not in VS Code core, thus forwarding to @weinand and @connor4312 for now

**Comment by weinand** [comment]
A configurable process picker is an excellent extension opportunity.

**Comment by bergmeister** [comment]
@aprilandjan If you use `processName`: "ReplaceWithProcessName", then that works. The problem is only when there is more than one process with that name, in that case it gives an error. In my opinion it should just present the picker with the filter already set to `ReplaceWithProcessName`.

**Comment by aprilandjan** [comment]
@bergmeister Thanks for your opinion! 

I tried `processName` in my node project, but It seems that the vscode node debugger does not support this `processName`,  maybe it is because that different language debuggers have different configurations?

Back to the topic, what we want are actually the same: define a process filter pattern, and allow manually pick if multiple matched results are founded. Seems that different language debugger must do that separately. 

Besides, I found that https://github.com/Microsoft/vscode/issues/50378 might be helpful to implement this.
